server/metrics/metrics.go

```diff
@@ -309,6 +313,37 @@ func newCertReloader(doneChan chan struct{}, certPath, keyPath string) (*certRel
 		keyPath:  keyPath,
 	}
 
+	// Generate self-signed certificate and key if the provided ones are not
+	// available.
+	_, errCertPath := os.Stat(certPath)
+	_, errKeyPath := os.Stat(keyPath)
+	if errCertPath != nil && os.IsNotExist(errCertPath) &&
+		errKeyPath != nil && os.IsNotExist(errKeyPath) {
+		logrus.Info("Metrics key and cert path does not exist, generating self-signed")
```

Comment:
Do you need to worry about the case when you have the certpath but not the key, or vice versa?

Reply:
Right now it only serves as HTTPs endpoint when both are given as string. If one of both does not exist on disk then we error as usual. I think all cases are already covered.